### ISSUE TITLE
[Feature] Custom Bio Fields

### ISSUE THREAD

**Issue by theotheroracle** [go-gitea/gitea#16244]
The bio field seems to be a bit small, and a bit unorganized . This isn't just a criticism of gitea, but all forges . 

The bio field is only a small text section with little context . Whilst this approach is flexible, it's not especially readable, and it's not very useful for multiple types of data . 

I propose a change similar to that of fediverse services, where there is a table of description and text for multipurpose data, which does see considerable use . 

This feature is so versatile and organized, i think it should be standard in more places . It's often used for pronouns ( which would address #13990 ), or links to other profiles, or other relevant information .

It's typically limited to 3 fields, although some servers allow more fields, and i think on gitea the number of fields should also be customizable .

Theoretically all bio fields could simply be customizable in this way, perhaps even allowing icons to be used instead of text descriptions if the user wanted .

**Comment by delvh** [comment]
I can see your point but I have to remark one thing:
> Theoretically all bio fields could simply be customizable in this way, perhaps even allowing icons to be used instead of text descriptions if the user wanted .

While that may sound appealing to users, this is a security nightmare as you give users the option to upload custom content that can be seen by others. That is in terms of security one of the worst things you can do to compromise every other security precaution taken, directly after calling  `eval()` on user generated input.
It is as far as I know pretty hard, but not impossible, to create a virus that expands to every user who looks at your profile by using this approach.

**Comment by theotheroracle** [comment]
oh no, i wasn't suggesting uploading custom icons, i was suggesting allowing picking from the icon set that gitea already uses .

**Comment by 6543** [comment]
most do not notice: bio-field/user-description do **support markdown**

it might be bit inconvenient if you dont have a whole form for bio but we are no job-scouting/dating/... platform ;)

an example how to customize:
![Bildschirmfoto zu 2021-06-25 17-23-50](https://user-images.githubusercontent.com/24977596/123447484-34c75e80-d5da-11eb-954b-fd90db373821.png)
-> *https://try.gitea.io/6543-TEST*

**Comment by 6543** [comment]
yes the 255 characters limit is a bit annoying ... but should work out in most cases

**Comment by zeripath** [comment]
Going back to the idea of a user backed repository - we could have these sections coming from user/user.git main BIOGRAPHY.md or so on

**Comment by 6543** [comment]
This is something more akceptable, there should be already a issue for it
